### DISCUSSION TITLE
npx tailwindcss takes several minutes to run when offline

### DISCUSSION THREAD

**Discussion by reubenfirmin** [tailwindlabs/tailwindcss#13100] [Help]
When I'm offline and run `npx tailwindcss`, the command hangs for approximately 2 minutes, before something times out and then the css is generated. I don't reference any remote stylesheets in my configuration. What's the cause of this, and can it be worked around?

**Comment by alex-krasikau** [comment]
Hey @reubenfirmin!

The delay you're experiencing when running ﻿npx tailwindcss in offline mode could be because ﻿npx is trying to fetch the latest version of tailwindcss from the npm registry before it times out. While you're online, this happens quite fast which is why you don't notice it.
To work around this, you can install tailwindcss locally in your project using npm or yarn. This way, ﻿npx will use the local version immediately, without trying to fetch it from the web.

Here's how you can do that with npm:
1. Install tailwindcss in your project:
```npm install tailwindcss```
2. Then you can use ﻿npx to run the local version:
```npx tailwindcss```

**Comment by reubenfirmin** [comment]
Good intuition @alex-krasikau  - that was the answer. I'm doing this within a kotlinjs project, and I _am_ installing tailwindcss via gradle+npm. However, it's clearly not ending up in the location that npx wants it to be.

What does npx fall back on in my initial scenario? Is it finding an old system-wide version or something?
